Library/ProcessOutputLines.Unix.cs

```diff
@@ -1,17 +1,254 @@
 ﻿using Microsoft.Win32.SafeHandles;
+using System.Buffers;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
 using System.Text;
 
 namespace System.TBA;
 
 public partial class ProcessOutputLines : IAsyncEnumerable<ProcessOutputLine>, IEnumerable<ProcessOutputLine>
 {
+    // P/Invoke declarations for poll
+    [StructLayout(LayoutKind.Sequential)]
+    private struct PollFd
+    {
+        public int fd;
+        public short events;
+        public short revents;
+    }
+
+    private const short POLLIN = 0x0001;
+    private const short POLLHUP = 0x0010;
+    private const short POLLERR = 0x0008;
+    private const int EINTR = 4; // Interrupted system call
+
+    [DllImport("libc", SetLastError = true)]
+    private static extern unsafe int poll(PollFd* fds, nuint nfds, int timeout);
+
+    [DllImport("libc", SetLastError = true)]
+    private static extern unsafe nint read(int fd, byte* buf, nuint count);
+
     public IEnumerator<ProcessOutputLine> GetEnumerator()
     {
         // NOTE: we could get current console Encoding here, it's omitted for the sake of simplicity of the proof of concept.
         Encoding encoding = _encoding ?? Encoding.UTF8;
         TimeoutHelper timeoutHelper = TimeoutHelper.Start(_timeout);
 
-        // The idea will be to start the proces as on Windows, but use `poll` or `epoll` to wait for data on the pipe fds.
-        yield break;
+        byte[] outputBuffer = ArrayPool<byte>.Shared.Rent(BufferHelper.InitialRentedBufferSize);
+        byte[] errorBuffer = ArrayPool<byte>.Shared.Rent(BufferHelper.InitialRentedBufferSize);
+        int outputStartIndex = 0, outputEndIndex = 0;
+        int errorStartIndex = 0, errorEndIndex = 0;
+
+        SafeFileHandle? parentOutputHandle = null, childOutputHandle = null, parentErrorHandle = null, childErrorHandle = null;
+        try
+        {
+            using SafeFileHandle inputHandle = Console.OpenStandardInputHandle();
+            File.CreateAnonymousPipe(out parentOutputHandle, out childOutputHandle);
+            File.CreateAnonymousPipe(out parentErrorHandle, out childErrorHandle);
+
+            using SafeProcessHandle processHandle = SafeProcessHandle.Start(_options, inputHandle, childOutputHandle, childErrorHandle);
+
+            _processId = processHandle.GetProcessId();
+
+            // Close child handles in parent process - they are now owned by the child
```

Comment:
The `Start` method takes care of that:

https://github.com/adamsitnik/ProcessPlayground/blob/0916209d4526e88be8e8163895dfa45a25351e9e/Library/SafeProcessHandle.cs#L35-L43